crates/aranya-client-capi/src/api/defs.rs

```diff
@@ -952,15 +997,207 @@ pub fn setup_default_roles(client: &mut Client, team: &TeamId) -> Result<(), imp
+/// The application can use `roles_len` to allocate a larger buffer.
+///
+/// @param[in] client the Aranya Client [`Client`].
+/// @param[in] team the team's ID [`TeamId`].
+/// @param[in] role the ID of the subject role [`Role`].
+/// @param[in] roles_out returns a list of roles that own `role` [`Role`].
+/// @param[in,out] roles_len the number of roles written to the buffer.
+pub fn role_owners(
+    client: &Client,
+    team: &TeamId,
+    role: &RoleId,
+    roles_out: *mut MaybeUninit<Role>,
+    roles_len: &mut usize,
+) -> Result<(), imp::Error> {
+    let owning_roles = client
+        .rt
+        .block_on(client.inner.team(team.into()).role_owners(role.into()))?
+        .__into_data();
+
+    if *roles_len < owning_roles.len() {
+        *roles_len = owning_roles.len();
+        return Err(imp::Error::BufferTooSmall);
+    }
+    let out = aranya_capi_core::try_as_mut_slice!(roles_out, *roles_len);
+
+    for (dst, src) in out.iter_mut().zip(owning_roles) {
+        Role::init(dst, imp::Role(src));
+    }
+
+    Ok(())
+}
+
+pub fn assign_role_management_permission(
+    client: &Client,
+    team: &TeamId,
+    role: &RoleId,
+    managing_role: &RoleId,
+    perm: Permission,
+) -> Result<(), imp::Error> {
+    // SAFETY: Caller must ensure `perm` is a valid C String.
+    let perm = unsafe { perm.as_underlying() }?;
+
+    client.rt.block_on(
+        client
+            .inner
+            .team(team.into())
+            .assign_role_management_permission(role.into(), managing_role.into(), perm),
+    )?;
+
+    Ok(())
+}
+
+pub fn revoke_role_management_permission(
+    client: &Client,
+    team: &TeamId,
+    role: &RoleId,
+    managing_role: &RoleId,
+    perm: Permission,
+) -> Result<(), imp::Error> {
+    // SAFETY: Caller must ensure `perm` is a valid C String.
+    let perm = unsafe { perm.as_underlying() }?;
+
+    client.rt.block_on(
+        client
+            .inner
+            .team(team.into())
+            .revoke_role_management_permission(role.into(), managing_role.into(), perm),
+    )?;
+
+    Ok(())
+}
+
+/// Returns all of the roles for this team.
+///
+/// Returns an `AranyaBufferTooSmall` error if the output buffer is too small to hold the roles.
+/// Writes the number of roles that would have been returned to `roles_len`.
+/// The application can use `roles_len` to allocate a larger buffer.
+///
+/// @param[in] client the Aranya Client [`Client`].
+/// @param[in] team the team's ID [`TeamId`].
+/// @param[in] roles_out returns a list of roles on the team [`Role`].
+/// @param[in,out] roles_len the number of roles written to the buffer.
+///
+/// @relates AranyaClient.
+pub fn roles(
+    client: &Client,
+    team: &TeamId,
+    roles_out: *mut MaybeUninit<Role>,
+    roles_len: &mut usize,
+) -> Result<(), imp::Error> {
+    let all_roles = client
+        .rt
+        .block_on(client.inner.team(team.into()).roles())?
+        .__into_data();
```

Comment:
nit: I don't see a reason to specify that this is all the roles. The `team.roles()` method doesn't.
```suggestion
    let roles = client
        .rt
        .block_on(client.inner.team(team.into()).roles())?
        .__into_data();
```